src/NetMQ/NetMQPoller.cs

```diff
@@ -120,25 +120,63 @@ protected override void QueueTask(Task task)
             m_tasksQueue.Enqueue(task);
         }
 
+        internal Task RunAsync([NotNull] Action action)
+        {
+            Task t;
+
+            if (!IsRunning || CanExecuteTaskInline)
+            {
+                action();
+
+                t = FromResult<object>(null);
+            }
+            else
+            {
+                t = Task.Factory.StartNew(action);
+            }
+
+            return t;
+        }
+
         /// <summary>
         /// Run an action on the Poller thread
         /// </summary>
         /// <param name="action">The action to run</param>
+        [Obsolete("Queues the action on the poller's thread, but provides no sync mechanism.  Please use RemoveAsync() to remove sockets or timers")]
```

Comment:
This message needs reviewing.

Reply:
I'm not sure we should force people to use the new async APIs. What do you think @somdoron?

Reply:
This doesn't force anyone to use the new API's, it just forces them to opt-out (with compiler ignores for release build).

It is strongly suggesting to use the new API's, but that is much better than the status-quo, IMHO: can't dispose sockets at all, just have to rely on GC.

Reply:
Is this a sticking point?  It can easily be removed, but is not my preference...  But need clear direction from NetMq team as to the final yay or nay.

What is the preferred method to communicate this issue and change to application coders?

Reply:
@drewnoakes So, I should interpret "I'm not sure..."  as "I'm not approving this until it removed" ?